torch/_subclasses/fake_tensor.py

```diff
@@ -803,6 +808,26 @@ class FakeTensor(torch.Tensor):
     fake_mode: "FakeTensorMode"
     constant: Optional[torch.Tensor]
 
+    # This memorizes the unbacked SymInt representing the number of nonzero
+    # elements in this tensor.  This is helpful if you do something like
+    # x[mask] and y[mask]; mask.nonzero() gets repeatedly called and should
+    # give a consistent unbacked SymInt.  It needs to be invalidated in the
+    # same way constant is.
+    # TODO: Generalize this as needed, e.g., into a trie of memos
```

Comment:
What would a trie of memos here look like ? For other operators besides nonzero ?

Reply:
Also if you repeatedly do a view on the tensor.